packages/amplify-graphql-transformer-migrator/src/__tests__/migration/migrate-tests.ts

```diff
@@ -0,0 +1,238 @@
+import { migrateGraphQLSchema } from '../../schema-migrator';
+import { parse } from 'graphql';
+
+describe('Schema migration tests', () => {
+  describe('Schema conversion tests', () => {
+    it('Should match expected output for base Todo and API key', () => {
+      const schema = /* graphql */
+      `type Todo @model {
+        id: ID!
+        name: String!
+        description: String
+      }
+      `;
+      const outputSchema = /* graphql */
+`type Todo @model @auth(rules: [{allow: public}]) {
+  id: ID!
+  name: String!
+  description: String
+}
+`;
+      const docNode = parse(schema);
+      const newSchema = migrateGraphQLSchema(schema, 'apiKey', docNode);
+      expect(newSchema).toMatch(outputSchema);
+    });
+
+    it('Should match expected output for extended Todo and API key', async () => {
+      const schema = /* graphql */
+      `type Todo @model {
+        id: ID!
+        name: String!
+        description: String
+      }
+      
+      type Ope @model {
+        foo: ID!
+        bar: String
+      }
+`;
+      const outputSchema = /* graphql */
+`type Todo @model @auth(rules: [{allow: public}]) {
+  id: ID!
+  name: String!
+  description: String
+}
+
+type Ope @model @auth(rules: [{allow: public}]) {
+  foo: ID!
+  bar: String
+}
+`;
+      const docNode = parse(schema);
+      const newSchema = migrateGraphQLSchema(schema, 'apiKey', docNode);
+      expect(newSchema).toMatch(outputSchema);
+    });
+
+    it('Should blow up if we try to use versioned', () => {
+      const schema = /* graphql */
+        `type Todo @model @versioned {
+        id: ID!
+        name: String!
+        description: String
+      }
+      
+      type Ope @model {
+        foo: ID!
+        bar: String
+      }
+`;
+
+      const docNode = parse(schema);
+      expect(() => migrateGraphQLSchema(schema, 'apiKey', docNode)).toThrowError();
+    });
+
+    it('Should migrate a primary key', () => {
```

Comment:
I think for most of these tests we can use [`test.each`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) to be more succinct

Reply:
Planning to make changes to the testing but I think we're going to ship this unhooked to the CLI flows first to unblock other testing